docs/contributor/03-11-hap-rules.md

```diff
@@ -0,0 +1,207 @@
+
+Rule Attributes are described in [Rule Attributes](#rule-attributes) section.
+
+## Rule Evaluation
+
+HAP evaluates a set of rules to determine what labels to use when querying secret bindings.
+Rule is evaluated during cluster provisioning.
+Rule entries are analyzed one by one.
+Physical attributes from every entry are compared with values from input, if they are the same the rule entry is considered triggered.
+If there is more than one triggered rule then only one is selected in accordance to priority rules described in the [Priority](#priority) section.
+If there are no rule entries trigered then the error is returned. No fallback behaviour is defined in such case.
+
+## Priority
+
+Only one rule can be triggered. If more than one rule entry match the request than only one is selected by sorting them by the number of attributes they contain.
+The entry that specifies most attributes (hence is more specific) is selected.
+If there are more than one rule with the same number of attributes then an error is returned.
+
+## Search Labels
+
+The HAP stores credentials for the hyperscaler accounts that have been set up in advance in Kubernetes Secrets that the Secret Bindings, that KEB searches for, point to. The SecretBindings are labeld with **hyperscalerType**, **shared** and **euAccess** labels. 
+
+The **hyperscaler-type** contains hyperscaler name and region information in the format `hyperscaler_type: <HYPERSCALER_NAME>[_<PLATFORM_REGION>][_<HYPERSCALER_REGION>]`, where both `_<PLATFORM_REGION>` and `_<HYPERSCALER_REGION>` are optional. The **hypercaler-type** label is mandatory. It is value is hardcoded and computed based on a plan in [hyperscaler_type.go](https://github.com/kyma-project/kyma-environment-broker/blob/main/common/hyperscaler/hyperscaler_type.go). This means that not all plans share their name with hyperscaler types, e.g. sap-converged-plan has `openstack` plan and `trial` plan can have either `azure` or `aws` trial depending on the cluster regions. The **euAccess** and **shared** labels contain boolean values and they used to divide existing pools to secrets used by EU restricted regions and secrets shared by multiple Global Accounts. The **euAccess** and **shared** labels are optional
+
+Every rule must contain at least a plan and apply `hyperscaler_type: <HYPERSCALER_NAME>` label. For example, the following example shows a simple rule entry configuration and a SecretBinding pool that this configuration corresponds to:
+
+```
+hap:
+  rule:
+    - gcp
+
+SecretBinding pools:
+- hyperscalerType: gcp
+```
+
+Rest of the examples in the document follow the same structure of presenting an example configuration and corresponding HAP pools as above snippet.
+
+## Rule Attributes
+
+Triggered rule entry applies that labels to the SecretBinding search query.
```

Comment:
Too many sentences about the same (IMHO).
Could we rephrase?

Reply:
applies = uses that labels during search?

Reply:
I would try to simplify this process (or description).
We do not create search and alter it later. 
We just create search labels based on matched rule and actual attributes.